### PR TITLE
Adjust the API in starttls plugin

### PR DESCRIPTION
**Pull request by yurynix**
Only that way I were able to get tls to work with
`"xmpp.js": "^0.2.0"`

### CODE REVIEW

**Comment by sonnyp** [comment]
@yurynix thanks

What server and/or domain?

**Comment by yurynix** [comment]
@sonnyp 
```js
client.start( { uri: 'xmpp://xmpp.wordpress.com', domain: 'im.wordpress.com' } );
```

I don't think it's related to the domain, but there is some kind of API mismatch with that plugin and the rest, the `socket` property is just not where it's expected to be...

File: packages/plugins/starttls/index.js [hunk 1]
```diff
@@ -13,26 +13,25 @@ const net = require('net')
 const NS = 'urn:ietf:params:xml:ns:xmpp-tls'
 
 function match(features, entity) {
-  return features.getChild('starttls', NS) &&
-    entity.socket.Socket === net.Socket // https://prosody.im/issues/issue/837
```
Comment (sonnyp) [inline packages/plugins/starttls/index.js]: please fix and leave the net.Socket check

File: packages/plugins/starttls/index.js [hunk 2]
```diff
@@ -13,26 +13,25 @@ const net = require('net')
...
-    options.socket = entity.socket.socket
-    entity.socket._detachSocket()
+    options.socket = entity.socket
+    entity._detachSocket()
     const tlsSocket = tls.connect(options, err => {
       if (err) {
         return reject(err)
       }
-      entity.socket._attachSocket(tlsSocket)
+      entity._attachSocket( tlsSocket )
```
Comment (sonnyp) [inline packages/plugins/starttls/index.js]: remove the surrounding spaces

File: packages/plugins/starttls/index.js (hunk 1 above)
Comment (yurynix) [inline packages/plugins/starttls/index.js]: I'm not really sure what this check trying to accomplish, as `entity.socket.Socket` is `undefined` always when I'm tested.

`entity.socket` is an instance of `net.Socket`, so we might want to ask:
`entity.socket instanceof net.Socket` ( will be also `true` for `TLSSocket` ) or `entity.socket.constructor ===  net.Socket` ( only `true` for `net.Socket` )
( i'll commit the later, as it seems to be most relevant to the original check )

**Pull request merged**